tests/integration/test_groupby.py

```diff
@@ -397,10 +397,48 @@ def test_groupby_split_every(c, gpu, split_every, expected_keys):
         input_ddf.groupby(by="user_id")
         .agg({"b": "sum"}, split_every=split_every)
         .reset_index(drop=False)
+        .rename(columns={"b": "S"})
+        .sort_values("user_id")
     )
-    expected_df = expected_df.rename(columns={"b": "S"})
-    expected_df = expected_df.sort_values("user_id")
+
     assert len(df.dask.keys()) == expected_keys
-    dd.assert_eq(df.compute().sort_values("user_id"), expected_df, check_index=False)
+    dd.assert_eq(df, expected_df, check_index=False)
+
+    c.drop_config("dask.groupby.aggregate.split_every")
+    c.drop_table("split_every_input")
+
+
+@pytest.mark.gpu
+@pytest.mark.parametrize("split_every,expected_keys", [(2, 91), (3, 85), (4, 81)])
+def test_groupby_split_every_gpu(c, split_every, expected_keys):
```

Comment:
Now that we are able to access dask-cudf's groupby agg codepath, the task graph size varies depending on if we use GPU or not. There could be a clean way to parametrize this all into one test, but I opted for this for now to make the difference more clear.

In general, is there a way we can verify that `split_every` is working that doesn't depend on checking the task graph? Asking because now upstream changes in dask-cudf can break this test.